app/views/facility_estimates/_estimate_detail.html.haml

```diff
@@ -1,9 +1,16 @@
 %tr{ data: { estimate_detail_id: estimate_detail.id } }
-  - id_prefix = "estimate[estimate_details_attributes][#{temp_id}]"
+  - numeric_id = defined?(index) ? index : rand(100000)
+  - id_prefix = "estimate[estimate_details_attributes][#{numeric_id}]"
+  - if estimate_detail.id.present?
+    = hidden_field_tag "#{id_prefix}[id]", estimate_detail.id
   = hidden_field_tag "#{id_prefix}[product_id]", estimate_detail.product_id
   - if estimate_detail.duration_unit.present?
     = hidden_field_tag "#{id_prefix}[duration_unit]", estimate_detail.duration_unit
-  = hidden_field_tag "#{id_prefix}[_destroy]", false, class: 'destroy-field'
+  
+  - if estimate_detail.price_policy_id.present?
```

Comment:
Do we need this `if`? The way I see it, all existing EstimateDetail will have a `price_policy_id` (it's a validation). If it doesn't exist, then the `price_policy_id` will be `nil` which is also fine